website/tools/minimize-language.js

```diff
@@ -5,9 +5,43 @@ import {
   existsSync,
 } from 'fs';
 
-for (const folder of readdirSync('./dist/locales', 'utf8')) {
-    if (existsSync('./dist/locales/' + folder + '/translation.json')) {
-      const content = readFileSync('./dist/locales/' + folder + '/translation.json', 'utf8',);
-      writeFileSync('./dist/locales/' + folder + '/translation.json', JSON.stringify(JSON.parse(content,),),);
+import yaml from 'yaml';
+import { dirname } from 'path';
+
+const generateJSONFromYAML = (yamlPath, outputPath) => {
+  if (existsSync(yamlPath)) {
+    try {
+      const content = readFileSync(yamlPath, 'utf8');
+      const jsonData = yaml.parse(content);
+      writeFileSync(outputPath, JSON.stringify(jsonData));
+    } catch {
+      console.log(`Error processing: ${yamlPath}`, error);
```

Comment:
the process has to exit with an error exit code when it fails - otherwise we might build incomplete websites

Reply:
I've heard the process.exit() is not the bestest practice? Or is it okay in that case? Should I just throw an error?

Reply:
Just not catching the error would be enough